File: app/src/main/java/com/xposed/miuiime/PropertyUtils.java
package com.xposed.miuiime;

import android.annotation.SuppressLint;

import java.lang.reflect.Method;

import de.robv.android.xposed.XposedBridge;

public class PropertyUtils {

    private static volatile Method get = null;

    @SuppressLint("PrivateApi")
    public static String get(String prop, String defaultvalue) {
        String value = defaultvalue;
        try {
            if (null == get) {
                synchronized (PropertyUtils.class) {
                    if (null == get) {
                        @SuppressLint("PrivateApi") Class<?> cls = Class.forName("android.os.SystemProperties");
                        get = cls.getDeclaredMethod("get", String.class, String.class);
                    }
                }
            }
            value = (String) (get.invoke(null, new Object[]{prop, defaultvalue}));
        } catch (Exception e) {
            XposedBridge.log(e);
        }
        return value;
    }
}

File: app/src/main/java/com/xposed/miuiime/PropertyUtils.kt
package com.xposed.miuiime

import android.annotation.SuppressLint
import de.robv.android.xposed.XposedBridge
import java.lang.reflect.Method

object PropertyUtils {
    private var get: Method? = null

    operator fun get(prop: String?, defaultValue: String?): String? {
        var value = defaultValue
        try {
            if (null == get) {
                synchronized(PropertyUtils::class.java) {
                    if (null == get) {
                        @SuppressLint("PrivateApi") val cls =
                            Class.forName("android.os.SystemProperties")
                        get = cls.getDeclaredMethod("get", String::class.java, String::class.java)
                    }
                }
            }
            value = get!!.invoke(null, prop, defaultValue) as String
        } catch (e: Exception) {
            XposedBridge.log(e)
        }
        return value
    }
}